### ISSUE TITLE
Effect run choppy, but in test mode they are smooth

### ISSUE THREAD

**Issue by TheAndi** [hyperion-project/hyperion.ng#648]
Hi again,

I discovered another problem on my RPi 1 B+ (running Raspbian Buster Lite).

When I run any dynamic effect on Hyperion the output is stuttering a lot. I thought it's my slow Pi 1, so I checked the CPU usage, but it's at ~40%.

And what is very interesting. The transitions between effects and colors are perfectly smooth.
Furthermore, when I load an effect into the Effects Configurator and hit "start test" all the effects run smooth as well except the some flicker mentioned in another issue. So the Pi 1 is totally capable of displaying them.

Is there a way this could be fixed?

Thanks in advance!

**Comment by mossieuparfait** [comment]
same here with a RPI 3B+.

**Comment by Joeboyc2** [comment]
If you disable smoothing do they run ok?

**Comment by Lord-Grey** [comment]
@mossieuparfait  Considering Smoothing is a good direction.
I am currently reviewing the smoothing code and  it seems that during testing an effect the standard smoothing parameters are used rather than those from the effect itself. 
In addition, if you edit an effect‘s smoothing parameters they will only be used after restart, as the effects‘ parameters are not updated correctly.

To check, if smoothing is the root cause, may I ask you to test a) with smoothing disabled as suggested by @Joeboyc2 b) configure the choppy effect with the general smoothing values (those which are used for the standard output) and check how the effect behaves after hyperion‘s restart

**Comment by TheAndi** [comment]
I tried disabling the global smoothing, but as mentioned in https://github.com/hyperion-project/hyperion.ng/issues/671 this does not work.

I also tried to create custom effects based on the given ones. When I just save them without a change they run choppy (as expected), but when I turn on smoothing in the effect setting, they run smooth. Unfortunately effects like "night rider" don't have a smoothing option, so no smooth version can be created.

I hope this helps to find the problem.
